src/sentry/snuba/metrics.py

```diff
@@ -447,16 +447,40 @@ class SnubaField:
         "operations": _AVAILABLE_OPERATIONS["metrics_sets"],
         "tags": _SESSION_TAGS,
     },
+    "transaction.duration": {
+        "type": "distribution",
+        "operations": _AVAILABLE_OPERATIONS["metrics_distributions"],
+        "tags": {
+            **_MEASUREMENT_TAGS,
+            "transaction.status": [
+                # Subset of possible states:
+                # https://develop.sentry.dev/sdk/event-payloads/transaction/
+                "ok",
+                "cancelled",
+                "aborted",
```

Comment:
`count(transaction.duration)` grouped by `transaction.status` can be used to determine failure rate.